app/controllers/application_controller.rb

```diff
@@ -14,6 +15,12 @@ def with_current_request
     end
   end
 
+  def check_entitlement
+    ManageIQ::API::Common::Request.with_request(request) do |current|
+      unauthorized_error(StandardError.new("User not Entitled")) unless current.entitlement.hybrid_cloud?
```

Comment:
Could we move this up into `with_current_request?` Then we don't need the extra callback and method.

Reply:
Sure I'm down with that. Updated to throw `Catalog::NotAuthorized` unless it has that entitlement.